SimPPS/RPDigiProducer/src/RPDigiProducer.cc

```diff
@@ -0,0 +1,184 @@
+
+    RP_hit_containers_.clear();
+    RP_hit_containers_ = conf.getParameter<std::vector<std::string> > ("ROUList");
+    verbosity_ = conf.getParameter<int> ("RPVerbosity");
+
+    simulateDeadChannels = false;
+    if (conf.exists("simulateDeadChannels")) { //check if "simulateDeadChannels" variable is defined in configuration file
+      simulateDeadChannels = conf.getParameter<bool> ("simulateDeadChannels");
+    }
+  }
+
+RPDigiProducer::~RPDigiProducer() {
+
+  // do anything here that needs to be done at desctruction time
+  // (e.g. close files, deallocate resources etc.)
+
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void RPDigiProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+
+  // initialize random engine
+  if(!rndEngine) {
+    Service<RandomNumberGenerator> rng;
+    if (!rng.isAvailable()) {
+      throw cms::Exception("Configuration") << "This class requires the RandomNumberGeneratorService\n"
+	"which is not present in the configuration file.  You must add the service\n"
+	"in the configuration file or remove the modules that require it.";
+    }
+    rndEngine = &(rng->getEngine(iEvent.streamID()));
+  }
+
+  // Step A: Get Inputs
+  edm::Handle<CrossingFrame<PSimHit> > cf;
+  //	iEvent.getByToken(tokenCrossingFrameTotemRP, cf);
+  iEvent.getByLabel("mix", "g4SimHitsTotemHitsRP", cf);
+
+  if (verbosity_) {
+    edm::LogInfo("RPDigiProducer") << "\n\n=================== Starting SimHit access" << "  ===================" << "\n";
+
+    std::auto_ptr<MixCollection<PSimHit> > col(
+	new MixCollection<PSimHit> (cf.product(), std::pair<int, int>(-0, 0)));
+    edm::LogInfo("RPDigiProducer") << *(col.get()) << "\n";
+    MixCollection<PSimHit>::iterator cfi;
+    int count = 0;
+    for (cfi = col->begin(); cfi != col->end(); cfi++) {
+      edm::LogInfo("RPDigiProducer") << " Hit " << count << " has tof " << cfi->timeOfFlight() << " trackid "
+	<< cfi->trackId() << " bunchcr " << cfi.bunch() << " trigger " << cfi.getTrigger()
+	<< ", from EncodedEventId: " << cfi->eventId().bunchCrossing() << " "
+	<< cfi->eventId().event() << " bcr from MixCol " << cfi.bunch() << "\n";
+      edm::LogInfo("RPDigiProducer") << " Hit: " << (*cfi) << "\n";
+      count++;
+    }
+  }
+
+  std::auto_ptr<MixCollection<PSimHit> > allRPHits(
+      new MixCollection<PSimHit> (cf.product(), std::pair<int, int>(0, 0)));
+
+  if (verbosity_)
+    edm::LogInfo("RPDigiProducer") << "Input MixCollection size = " << allRPHits->size() << "\n";
+
+  //Loop on PSimHit
+  SimHitMap.clear();
+
+  MixCollection<PSimHit>::iterator isim;
+  for (isim = allRPHits->begin(); isim != allRPHits->end(); ++isim) {
+    SimHitMap[(*isim).detUnitId()].push_back((*isim));
+
+  }
+
+  // Step B: LOOP on hits in event
+  theDigiVector.reserve(400);
+  theDigiVector.clear();
```

Comment:
`theDigiVector` should be a local variable (without `the` to not confuse with one convention of member variables).